Python/Product/Analysis/EncodedLocation.cs

```diff
@@ -39,16 +39,16 @@ struct EncodedLocation : IEquatable<EncodedLocation> {
         public readonly object Location;
 
         public EncodedLocation(ILocationResolver resolver, object location) {
+            if (resolver == null && !(location is LocationInfo)) {
+                throw new ArgumentNullException(nameof(resolver));
+            }
+
             Resolver = resolver;
             Location = location;
         }
 
         public override int GetHashCode() {
-            if (Location != null) {
-                return Resolver.GetHashCode() ^ Location.GetHashCode();
-            }
-
-            return Resolver.GetHashCode();
+            return (Resolver?.GetHashCode() ?? 0) ^ (Location?.GetHashCode() ?? 0);
```

Comment:
Ah, I forgot about that! It depends on a NuGet package that provides `System.ValueTuple` (the lightbulb on the error should say as much). The other option is to target .NET 4.7, which has it in the box; but I'm not sure whether VS itself has that as hard dependency, and hence whether we can target it or not.

The NuGet package is available for all .NET targets. This is going to be the general approach for new C# features requiring additional types going forward (they also did it with `INotifyCompletion`, for example).